db/fixtures/ae_datastore/Deployment/System/StateMachine/Deployment.class/__methods__/create_ansible_inventory.rb

```diff
@@ -0,0 +1,119 @@
+require 'net/ssh'
+
+INVENTORY_FILE = 'inventory.yaml'.freeze
+RHEL_SUBSCRIBE_INVENTORY = 'rhel_subscribe_inventory.yaml'.freeze
+
+def create_inventory_template(deployment_master, masters, nodes, user)
+  template = "
+[OSEv3:children]
+masters
+nodes
+
+[masters:vars]
+ansible_ssh_user=#{user}
+ansible_sudo=true
+deployment_type=openshift-enterprise
+#openshift_use_manageiq=True
+
+[nodes:vars]
+ansible_ssh_user=#{user}
+ansible_sudo=true
+deployment_type=openshift-enterprise
+#openshift_use_manageiq=True
+
+[masters]
+#{deployment_master} ansible_connection=local openshift_scheduleable=True
+#{masters.join("\n") unless masters.empty?}
+
+[nodes]
+#{nodes.join("\n")}
+  "
+
+  template
+end
+
+def create_rhel_subscribe_template(masters, nodes, user)
+  template = "
+[OSEv3:children]
+masters
+nodes
+
+[OSEv3:vars]
+ansible_ssh_user=#{user}
+
+deployment_type=openshift-enterprise
+
+rhsub_user=#{$evm.root['rhsub_user']}
+rhsub_pass=#{$evm.root['rhsub_pass']}
+# This is defaulted to 'OpenShift Enterprise, Premium*'
+rhsub_pool=#{$evm.root['rhsub_pool']}
+# Optional server
+# rhsub_server=subscription.rhn.stage.redhat.com
+
+[masters]
+#{masters.join("\n") unless masters.empty?}
+
+[nodes]
+#{nodes.join("\n")}
+  "
+  template
+end
+
+def replace_connecting_master_ip(nodes, master_ip)
+  nodes.each_with_index do |node, index|
+    if node.include? master_ip
+      nodes[index] = "#{master_ip}              ansible_connection=local"
+    end
+  end
+  nodes
+end
+
+def remove_deployment_master(hosts, master_ip)
+  hosts.each do |host|
+    if host.include? master_ip
+      hosts.delete(host)
+    end
+  end
+  hosts
+end
+
+def create_ansible_inventory_file(subscribe = false)
+  if subscribe
+    $evm.log(:info, "creating #{RHEL_SUBSCRIBE_INVENTORY}")
+    masters = remove_deployment_master($evm.root['masters'], $evm.root['deployment_master'])
```

Comment:
remove_deployment_master can be replaced by:
```
$evm.root['masters'].reject { |host| host.include?($evm.root['deployment_master']) }
```
.select { |host| !host.include?("B") }